test/test_integration.py

```diff
@@ -672,6 +673,98 @@ def socket_handler(listener):
         recv_event.set()
         self.tear_down()
 
+    def test_secure_proxy_connection(self):
+        self.set_up(secure=True, secure_auto_wrap_socket=False, proxy=True)
+
+        data = []
+        req_event = threading.Event()
+        recv_event = threading.Event()
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            # Read the CONNECT reader
```

Comment:
Aww, what a shameful copy-pasted typo 😃

BTW, CONNECT is a method, [according to RFC 7231 HTTP/1.1](https://tools.ietf.org/html/rfc7231#section-4.3.6)
I will look through my comments again, thanks!